Sources/HummingbirdBasicAuth/PasswordVerifier.swift

```diff
@@ -0,0 +1,33 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Bcrypt
+import NIOPosix
+
+/// Protocol for password verifier
+public protocol PasswordVerifier: Sendable {
+    func verifyPassword(_ text: String, hash: String) async throws -> Bool
```

Comment:
```suggestion
    func verifyPassword(_ password: String, createsHash hash: String) async throws -> Bool
```

I would love an argument name of some form to clarify to newcomers what the hash's relation is to the passwod